### PR TITLE
feat(dataarts): support data connection resource

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
support the new data connection resource

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new resource support: data connection.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dataarts' TESTARGS='-run=TestAccResourceDataConnection_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dataarts -v -run=TestAccResourceDataConnection_basic -timeout 360m -parallel 4
=== RUN   TestAccResourceDataConnection_basic
=== PAUSE TestAccResourceDataConnection_basic
=== CONT  TestAccResourceDataConnection_basic
--- PASS: TestAccResourceDataConnection_basic (38.31s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dataarts  38.393s
```

### CODE REVIEW

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_data_connection.go [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+				Description: `The data connection type.`,
+			},
+			"config": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringIsJSON,
+				DiffSuppressFunc: func(_, o, n string, _ *schema.ResourceData) bool {
+					oldRaws := unmarshalConfigContent(o)
+					newRaws := unmarshalConfigContent(n)
+					// The API response will return a attribute 'update_time', which is a random string generated by the create
```
Comment (houpeng80) [inline huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_data_connection.go]:
```suggestion
					// The API response will return an attribute 'update_time', which is a random string generated by the create
```

File: huaweicloud/services/acceptance/dataarts/resource_huaweicloud_dataarts_studio_data_connection_test.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+					resource.TestCheckResourceAttr(resourceName, "name", updateName),
+					resource.TestCheckResourceAttr(resourceName, "env_type", "0"),
+					resource.TestCheckResourceAttrSet(resourceName, "config"),
+				),
+			},
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"tags", "period_unit", "period", "auto_renew"},
```
Comment (houpeng80) [inline huaweicloud/services/acceptance/dataarts/resource_huaweicloud_dataarts_studio_data_connection_test.go]: it is useless

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_data_connection.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_data_connection.go]: done.

File: huaweicloud/services/acceptance/dataarts/resource_huaweicloud_dataarts_studio_data_connection_test.go (hunk 2 above)
Comment (Lance52259) [reply huaweicloud/services/acceptance/dataarts/resource_huaweicloud_dataarts_studio_data_connection_test.go]: yeah

**Pull request merged**